sljit_src/sljitNativeARM_64.c

```diff
@@ -2509,18 +2532,68 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_load(struct sljit_compiler
 	}
 
 	return push_inst(compiler, ins | RN(mem_reg) | RT(dst_reg));
+#endif /* ARM_FEATURE_ATOMICS */
 }
 
 SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_store(struct sljit_compiler *compiler, sljit_s32 op,
 	sljit_s32 src_reg,
 	sljit_s32 mem_reg,
 	sljit_s32 temp_reg)
 {
-	sljit_ins ins;
+	sljit_ins ins, l;
+	sljit_ins cmp = 0;
+	sljit_sw m = 0;
+	sljit_s32 src = src_reg;
 
 	CHECK_ERROR();
 	CHECK(check_sljit_emit_atomic_store(compiler, op, src_reg, mem_reg, temp_reg));
 
+#ifdef __ARM_FEATURE_ATOMICS
+	if (op & SLJIT_SET_ATOMIC_STORED)
+		cmp = (SUBS ^ W_OP) | RD(TMP_ZERO);
+
+	switch (GET_OPCODE(op)) {
+	case SLJIT_MOV32:
+	case SLJIT_MOV_U32:
+		ins = CAS ^ (1 << 30);
+		l = LDRSW;
```

Comment:
Like all other operations, this will cause "Undefined Behaviour" if not correctly aligned, but would need special treatment to avoid that if the higher half or a worded size address is targeted.